### DISCUSSION TITLE
shell commands can break visuals

### DISCUSSION THREAD

**Discussion by rmburg** [helix-editor/helix#11314] [General]
### Summary

when running a shell command like `fzf` which clears the screen or does other things to the terminal, things break a bit:
- helix does not rerender after the shell command exits, causing leftover visuals from the shell command
- when exiting helix in this state, the last output from helix sticks around in the terminal 

### Reproduction Steps

1. open helix
2. run `:insert-output fzf`
3. close fzf
4. observe breakage
5. run `:redraw`
6. observe: everything looks normal again
7. quit helix
8. observe: output is not cleared as it should

### Helix log

(nothing is printed to the log)

### Platform

Arch Linux

### Terminal Emulator

kitty 0.35.2

### Installation Method

helix-git AUR package

### Helix Version

24.07.r59.g86795a9dc-1

**Comment by pascalkuthe** [comment]
invoking TUI programs from helix is not supported. All shell related functionality is only meant for non-interactive commands. Intreactive commands will only work once we have a builtin terminal emulator (and probably still not with `:insert-output`).
